src/main/java/uk/gov/companieshouse/company/metrics/config/ApplicationConfig.java

```diff
@@ -23,4 +28,24 @@ EnvironmentReader environmentReader() {
     public InternalApiClient internalApiClient() {
         return ApiSdkManager.getPrivateSDK();
     }
+
+    @Primary
+    @Bean
+    public ObjectMapper defaultObjectMapper() {
+        return new Jackson2ObjectMapperBuilder().build();
+    }
+
+    /**
+     * Create a custom mapper to use in Mongo reading/writing converters.
+     * @return a custom object mapper
+     */
+    @Bean("mongoConverterMapper")
+    public ObjectMapper customMapper() {
```

Comment:
I think this bean will override default Object mapper used by Spring to de-serialise json payloads, have you tested sending a json payload and see if its de-serialised correctly?

Reply:
have used the same converters as in charges/insolvency data api and added to this component also. had to also add json writer settings otherwise it was giving error while converting from bson to json